napari/utils/transforms/transform_utils.py

```diff
@@ -0,0 +1,225 @@
+            gamma = np.deg2rad(rotate[2])
+        else:
+            alpha = rotate[0]
+            beta = rotate[1]
+            gamma = rotate[2]
+        R_alpha = np.array(
+            [
+                [np.cos(alpha), np.sin(alpha), 0],
+                [-np.sin(alpha), np.cos(alpha), 0],
+                [0, 0, 1],
+            ]
+        )
+        R_beta = np.array(
+            [
+                [np.cos(beta), 0, np.sin(beta)],
+                [0, 1, 0],
+                [-np.sin(beta), 0, np.cos(beta)],
+            ]
+        )
+        R_gamma = np.array(
+            [
+                [1, 0, 0],
+                [0, np.cos(gamma), -np.sin(gamma)],
+                [0, np.sin(gamma), np.cos(gamma)],
+            ]
+        )
+        rotate_mat = R_alpha @ R_beta @ R_gamma
+    else:
+        # Otherwise assume a full nD rotation matrix has been passed
+        rotate_mat = np.array(rotate)
+    n_rotate = rotate_mat.shape[0]
+
+    # Convert a scale vector to an nD diagonal matrix
+    scale_mat = np.diag(scale)
+    n_scale = scale_mat.shape[0]
+
+    # Check if an upper-triangular representation of shear or
+    # a full nD shear matrix has been passed
+    if np.isscalar(shear):
+        raise ValueError(
+            'Scalars are not valid values for shear.'
+            ' Shear must be an upper triangular vector'
+            ' or square matrix with ones along the main'
+            ' diagonal.'
+        )
+    if np.array(shear).ndim == 1:
+        shear_mat = expand_upper_triangular(shear)
+    else:
+        shear_mat = np.array(shear)
+
+    # Check the dimensionality of the transforms and pad as needed
+    n_shear = shear_mat.shape[0]
+    ndim = max(n_scale, n_rotate, n_shear)
+
+    full_scale = embed_in_identity_matrix(scale_mat, ndim)
+    full_rotate = embed_in_identity_matrix(rotate_mat, ndim)
+    full_shear = embed_in_identity_matrix(shear_mat, ndim)
+    return full_rotate @ full_scale @ full_shear
+
+
+def expand_upper_triangular(vector):
+    """Expand a vector into an upper triangular matrix.
+
+    Decomposition is based on code from https://github.com/matthew-brett/transforms3d.
+    In particular, the `striu2mat` function in the `shears` module.
+    https://github.com/matthew-brett/transforms3d/blob/0.3.1/transforms3d/shears.py#L30-L77.
+
+    Parameters
+    ----------
+    vector : np.array
+        1D vector of length M
+
+    Returns
+    -------
+    upper_tri : np.array shape (N, N)
+        Upper triangluar matrix.
+    """
+    n = len(vector)
+    N = ((-1 + np.sqrt(8 * n + 1)) / 2.0) + 1  # n+1 th root
+    if N != np.floor(N):
+        raise ValueError('%d is a strange number of shear elements' % n)
+    N = int(N)
+    inds = np.triu(np.ones((N, N)), 1).astype(bool)
+    upper_tri = np.eye(N)
+    upper_tri[inds] = vector
+    return upper_tri
+
+
+def embed_in_identity_matrix(matrix, ndim):
+    """Embed an MxM matrix in a larger NxN identity matrix.
```

Comment:
Specify *where* in the matrix it is embedded? bottom right from my reading of the code?

Reply:
correct, done